source/pervasive/map.rs

```diff
@@ -68,6 +70,24 @@ pub fn axiom_map_insert_different<K, V>(m: Map<K, V>, key1: K, key2: K, value: V
     ensures(equal(m.insert(key2, value).index(key1), m.index(key1)));
 }
 
+#[proof]
+#[verifier(external_body)]
+#[verifier(broadcast_forall)]
+pub fn axiom_map_remove_domain<K, V>(m: Map<K, V>, key: K) {
+    ensures(equal(#[trigger] m.remove(key).dom(), m.dom().remove(key)));
+}
+
+#[proof]
+#[verifier(external_body)]
+#[verifier(broadcast_forall)]
+pub fn axiom_map_remove_different<K, V>(m: Map<K, V>, key1: K, key2: K) {
+    requires([
+        m.dom().contains(key1),
```

Comment:
strictly speaking, we don't actually need this condition, do we?

do we have a philosophy on such things?

Reply:
I think we don't strictly need it for soundness.

I'm matching `axiom_map_insert_different` here; I believe the strategy has been to keep axioms just as powerful as they need to, and not more (at least until we have the option of multiple/a-la-carte `broadcast_forall` import sets). @Chris-Hawblitzel may have more to say.